README.md

```diff
@@ -0,0 +1,180 @@
+# graphql-node-resource
+
+This library implements the [Node GraphQL type defined by Relay](https://relay.dev/docs/guides/graphql-server-specification/#object-identification). Relay requires that a GraphQL server
+has a way to identify and refetch GraphQL resources. Its meant to be used within a GraphQL API that fetches its types from a REST API.
+
+
+```mermaid
+graph LR;
+    frontend(frontend using relay)-->graphql-api(GraphQL API);
+    graphql-api-->rest-api(REST API);
+    rest-api-->graphql-api;
+    graphql-api-->frontend;
+```
+_example architecture_
+
+
+## What is a Node?
+
+A `Node` is a refetchable GraphQL resource. All `Node`s must have a _globally unique ID_ and be fetchable via this ID. Relay will refetch these resources in the following way:
+
+```
+node(id: $resource_id) {
+    ... on Foo {
+        bar
+        baz
+    }
+}
+```
+
+## The Node Class
```

Comment:
question: this is kind of ambiguous to me, should I be reading this something like, "The NodeType, a sensible Node Implementation"?